vmdb/spec/controllers/miq_request_controller_spec.rb

```diff
@@ -18,4 +18,90 @@
+    context "MiqRequest-requester_id set based on user_choice" do
+      let(:path) { ["and", 2, "=", "value"] }
+
+      it "selected 'all'" do
+        MiqExpression.should_receive(:new).with { |h| expect(h.fetch_path(path) == "all").to be_false }
+        controller.send(:prov_condition, :user_choice => "all")
+      end
+
+      it "selected '1'" do
+        MiqExpression.should_receive(:new).with { |h| expect(h.fetch_path(path) == 1).to be_true }
+        controller.send(:prov_condition, :user_choice => 1)
+      end
+    end
+
+    it "MiqRequest-approval_state set with :applied_states" do
+      content = [{"=" => {"value" => "state", "field" => "MiqRequest-approval_state"}}, {"=" => {"value" => "state 2", "field" => "MiqRequest-approval_state"}}]
+      MiqExpression.should_receive(:new).with { |h| expect(h.fetch_path("and", 2, "or")).to eq(content) }
+      controller.send(:prov_condition, :applied_states => ["state", "state 2"])
+    end
+
+    it "MiqRequest-resource_type" do
+      content = [
+        {"=" => {"value" => "[:MiqProvisionRequest, {:template=>\"VM Provision\", :clone_to_vm=>\"VM Clone\", :clone_to_template=>\"VM Publish\"}]", "field" => "MiqRequest-resource_type"}},
+        {"=" => {"value" => "[:VmReconfigureRequest, {:vm_reconfigure=>\"VM Reconfigure\"}]", "field" => "MiqRequest-resource_type"}},
+        {"=" => {"value" => "[:VmMigrateRequest, {:vm_migrate=>\"VM Migrate\"}]", "field" => "MiqRequest-resource_type"}},
+        {"=" => {"value" => "[:ServiceTemplateProvisionRequest, {:clone_to_service=>\"Service Provision\"}]", "field" => "MiqRequest-resource_type"}},
+        {"=" => {"value" => "[:ServiceReconfigureRequest, {:service_reconfigure=>\"Service Reconfigure\"}]", "field" => "MiqRequest-resource_type"}},
+      ]
+      MiqExpression.should_receive(:new).with { |h| expect(h.fetch_path("and", 2, "or")).to eq(content) }
+      controller.send(:prov_condition, {})
+    end
+
+    context "MiqRequest-request_type set based on type_choice" do
+      let(:path) { ["and", 3, "=", "value"] }
+
+      it "selected 'all'" do
+        MiqExpression.should_receive(:new).with { |h| expect(h.fetch_path(path)).to be_nil }
+        controller.send(:prov_condition, :type_choice => "all")
+      end
+
+      it "selected '1'" do
+        MiqExpression.should_receive(:new).with { |h| expect(h.fetch_path(path)).to eq(1) }
+        controller.send(:prov_condition, :type_choice => 1)
+      end
+    end
+
+    it "MiqRequest-reason_text" do
+      content = {"value" => "just because", "field" => "MiqRequest-reason"}
+      MiqExpression.should_receive(:new).with { |h| expect(h.fetch_path("and", 3, "INCLUDES")).to eq(content) }
+      controller.send(:prov_condition, :reason_text => "just because")
+    end
+
+    it "empty options hash" do
+      MiqExpression.should_receive(:new).with do |h|
+        expect(h.fetch_path("and", 2, "or", 0, "=", "field") == "MiqRequest-approval_state").to be_false  # Doesn't set approval_states
```

Comment:
@Fryguy here we're looking for the negative case.  Unfortunately ```.to_not eq``` doesn't work here.